javatests/dagger/internal/codegen/GraphValidationTest.java

```diff
@@ -2071,6 +2072,142 @@ public void subcomponentBindingConflictsWithParent() {
         .onLine(8);
   }
 
+
+  @Test
+  public void subcomponentBindingConflictsWithParentWhenSubcomponentPerformsMembersInjection() {
+    JavaFileObject parent =
+        JavaFileObjects.forSourceLines(
+            "test.Parent",
+            "package test;",
+            "",
+            "import dagger.Component;",
+            "import dagger.Module;",
+            "import dagger.Provides;",
+            "",
+            "@Component(modules = Parent.ParentModule.class)",
+            "interface Parent {",
+            "  Child child();",
+            "",
+            "  @Module",
+            "  static class ParentModule {",
+            "    @Provides static Object object() {",
+            "      return \"parent\";",
+            "    }",
+            "  }",
+            "}");
+    JavaFileObject child =
+        JavaFileObjects.forSourceLines(
+            "test.Child",
+            "package test;",
+            "",
+            "import dagger.Module;",
+            "import dagger.Provides;",
+            "import dagger.Subcomponent;",
+            "",
+            "@Subcomponent(modules = Child.ChildModule.class)",
+            "interface Child {",
+            "  void inject(Injected injected);",
+            "",
+            "  @Module",
+            "  static class ChildModule {",
+            "    @Provides static Object object() {",
+            "      return \"child\";",
+            "    }",
+            "  }",
+            "}");
+    JavaFileObject injected =
+        JavaFileObjects.forSourceLines(
+            "test.Injected",
+            "package test;",
+            "",
+            "import javax.inject.Inject;",
+            "",
+            "final class Injected {",
+            "  @Inject Object object;",
+            "}");
+    assertAbout(javaSources())
```

Comment:
For new tests, please use the new Compile Testing APIs. See `Compilers.daggerCompiler()` in this package and how other tests use it.